params/precompile_upgrade.go

```diff
@@ -0,0 +1,261 @@
+// (c) 2022 Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package params
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/ava-labs/subnet-evm/precompile/config"
+	precompileConfig "github.com/ava-labs/subnet-evm/precompile/config"
+	"github.com/ava-labs/subnet-evm/precompile/modules"
+	"github.com/ava-labs/subnet-evm/utils"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+var errMultipleKeys = errors.New("PrecompileUpgrade must have exactly one key")
+var errNoKey = errors.New("PrecompileUpgrade cannot be empty")
+
+// PrecompileUpgrade is a helper struct embedded in UpgradeConfig.
+// It is used to unmarshal the json into the correct precompile config type
+// based on the key. Keys are defined in each precompile module, and registered in
+// params/precompile_modules.go.
+type PrecompileUpgrade struct {
+	config.Config
+}
+
+// UnmarshalJSON unmarshals the json into the correct precompile config type
+// based on the key. Keys are defined in each precompile module, and registered in
+// params/precompile_modules.go.
+// Ex: {"feeManagerConfig": {...}} where "feeManagerConfig" is the key
+func (u *PrecompileUpgrade) UnmarshalJSON(data []byte) error {
+	raw := make(map[string]json.RawMessage)
+	if err := json.Unmarshal(data, &raw); err != nil {
+		return err
+	}
+	if len(raw) == 0 {
+		return errNoKey
+	}
+	if len(raw) > 1 {
+		return errMultipleKeys
+	}
```

Comment:
Can we update this to a single formatted error that requires the length of raw is 1 and returns the actual number if anything different is found?

Reply:
done